File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio18  cuantos digitos tiene un numero entero positivo o negativo/archives/18.py
import sys
import math

# Obtener los argumentos de la línea de comandos
b = int(sys.argv[1])
a = abs(b)
if 0 <= a <= 99999:
    if 0 <= a <= 9:
        dig = 1
    elif 10 <= a <= 99:
        dig = 2
    elif 100 <= a <= 999:
        dig = 3
    elif 1000 <= a <= 9999:
        dig = 4
    elif 10000 <= a <= 99999:
        dig = 5
    mensaje = f"""El numero {b} tiene <br> {dig} {"digito" if dig==1 else "digitos"}"""
else:
    mensaje = "Ingrese un maximo de 5 digitos"
    

print(mensaje)

File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio18  cuantos digitos tiene un numero entero positivo o negativo/archives/18.php
<?php
$num1 = ($_POST["num1"]);
$num1= ltrim($num1, "0");
if (
	isset($num1) &&
	trim($num1) !== "" &&
	is_numeric($num1) &&
	filter_var($num1, FILTER_VALIDATE_INT) !== false
) {
	/* SECTION inicio programa */
	$a = abs($num1);
	$dig;
	if ($a >= 0 && $a <= 99999) {
		switch (true) {
			case ($a <= 9):
				$dig = 1;
				break;
			case ($a <= 99):
				$dig = 2;
				break;
			case ($a <= 999):
				$dig = 3;
				break;
			case ($a <= 9999):
				$dig = 4;
				break;
			case ($a <= 99999):
				$dig = 5;
				break;
		}
		$mensaje = "El numero $num1 tiene: <br> " . $dig . ($dig == 1 ? " digito" : " digitos");
	} else {
		$mensaje = "Solo se aceptan numeros de hasta 5 digitos ";
	}
	/* !SECTION fin programa */
} else {
	$mensaje = "Ingrese un número Entero.";
}
echo $mensaje;
